### PR TITLE
ssl: do not choose auto DH groups that are weaker than the security level

### PR DESCRIPTION
**Pull request by paulidale**
Manual merge of #15818 to 1.1.1.

- [ ] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

**Comment by paulidale** [comment]
Test issues.

File: test/recipes/80-test_ssl_old.t [hunk 1]
```diff
@@ -476,18 +476,21 @@ sub testssl {
...
 
 	SKIP: {
-	    skip "skipping anonymous DH tests", 1
+	    skip "skipping anonymous DH tests", 2
 	      if ($no_dh);
 
 	    ok(run(test([@ssltest, "-v", "-bio_pair", "-tls1", "-cipher", "ADH", "-dhe1024dsa", "-num", "10", "-f", "-time"])),
 	       'test tlsv1 with 1024bit anonymous DH, multiple handshakes');
+
+	    ok(run(test(['ssltest_old', '-psk', '0102030405', '-cipher', '@SECLEVEL=2:DHE-PSK-AES128-CCM'])),
```
Comment (slontis) [inline test/recipes/80-test_ssl_old.t:492]: This will have the same issue as the other PR.. i.e it needs a seperate skip section

Comment (paulidale) [reply test/recipes/80-test_ssl_old.t:492]: I made the change here first, tests break...  Still digging...

**Comment by paulidale** [comment]
The test failure is related to SSLv3 and key sizes.

**Comment by paulidale** [comment]
Reviewable now.

**Review by t8m — APPROVED** [review]

**Comment by paulidale** [comment]
Merged.

**Pull request closed without merging**